File: src_backup/main/java/com/freehands/assistant/BootReceiver.java
package com.freehands.assistant;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = "BootReceiver";
    
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        
        if (Intent.ACTION_BOOT_COMPLETED.equals(action) ||
            Intent.ACTION_MY_PACKAGE_REPLACED.equals(action) ||
            Intent.ACTION_PACKAGE_REPLACED.equals(action)) {
            
            if (BuildConfig.DEBUG) { Log.d(TAG, "Boot completed or package updated, starting FreeHands service"); }

            
            try {
                // Start the voice listening service
                Intent serviceIntent = new Intent(context, VoiceListeningService.class);
                context.startForegroundService(serviceIntent);
                
                if (BuildConfig.DEBUG) { Log.d(TAG, "FreeHands service started successfully"); }

                
            } catch (Exception e) {
                Log.e(TAG, "Error starting FreeHands service on boot", e);
            }
        }
    }
}

File: src_backup/main/java/com/freehands/assistant/BootReceiver.kt
package com.freehands.assistant

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import timber.log.Timber

/**
 * BroadcastReceiver that starts the VoiceListeningService when the device boots up.
 * This ensures that the voice assistant is always running in the background.
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent?) {
        if (intent?.action == Intent.ACTION_BOOT_COMPLETED || 
            intent?.action == Intent.ACTION_MY_PACKAGE_REPLACED) {
            
            Timber.d("Boot completed or app updated, starting voice service")
            
            // Start the voice listening service
            val serviceIntent = Intent(context, VoiceListeningService::class.java)
            
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(serviceIntent)
            } else {
                context.startService(serviceIntent)
            }
        }
    }
}
